src/module-magespices-mage2dark/Plugin/Model/View/Design.php

```diff
@@ -0,0 +1,61 @@
+<?php
+namespace MageSpices\Mage2Dark\Plugin\Model\View;
+
+use Magento\Framework\App\Area;
+use Magento\Framework\App\Config\ScopeConfigInterface;
+use Magento\Framework\Stdlib\DateTime\TimezoneInterface;
+
+class Design
+{
+    protected $scopeConfig;
+
+    protected $timeZone;
+
+    protected $themeSelectionConfig;
+
+    public function __construct(
+        ScopeConfigInterface $scopeConfig,
+        TimezoneInterface $timezone
+    ) {
+        $this->scopeConfig = $scopeConfig;
+        $this->timeZone = $timezone;
+    }
+
+    /**
+     * @param \Magento\Theme\Model\View\Design $subject
+     * @param callable $proceed
+     * @param string|null $area
+     * @param array $params
+     * @return
+     */
+    public function aroundGetConfigurationDesignTheme(
+        \Magento\Theme\Model\View\Design $subject,
+        callable $proceed,
+        $area = null,
+        array $params = []
+    ) {
+        if (!$area) {
+            $area = $subject->getArea();
+        }
+        if ($area == Area::AREA_ADMINHTML && $this->getThemeSelectionConfig() == "dark") {
+            return "MageSpices/Mage2Dark";
+        }
+        return $proceed($area, $params);
+    }
+
+    protected function getThemeSelectionConfig()
+    {
+        if (!$this->themeSelectionConfig) {
+            $this->themeSelectionConfig = $this->scopeConfig->getValue('mage2dark/settings/theme_selection');
+            if ($this->themeSelectionConfig == "auto") {
+                $currentHour = $this->timeZone->date()->format("H");
+                if (($currentHour >= 19 || $currentHour < 7)) {
```

Comment:
You can replace 19 and 7 (which is actually magic number in this case) and replace with private constants:
`HOUR_NIGHT_MODE_ON`
`HOUR_NIGHT_MODE_OFF`
Or something of this type :-)

Reply:
Actually, adding an admin configuration to choose the time slot for dark theme instead of fixed values in the code. will push it soon.

Reply:
Added a new setting for time slot selection.

![config](https://user-images.githubusercontent.com/5336201/70606044-ad95de00-1c21-11ea-8bc3-b747b30f7a49.png)